chart/templates/configmap.yaml

```diff
@@ -0,0 +1,117 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+################################
+## Airflow ConfigMap
+#################################
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "airflow_config" . }}
+  labels:
+    tier: airflow
+    component: config
+    release: {{ .Release.Name }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    heritage: {{ .Release.Service }}
+{{- with .Values.labels }}
+{{ toYaml . | indent 4 }}
+{{- end }}
+data:
+  # These are system-specified config overrides.
+  airflow.cfg: |
+    [core]
+    load_examples = False
+    colored_console_log = False
+    executor = {{ .Values.executor }}
+{{- if .Values.elasticsearch.enabled }}
+    remote_logging = True
+{{- end }}
+
+    [webserver]
+    enable_proxy_fix = True
+    expose_config = True
+    rbac = True
+
+{{- if eq .Values.executor "CeleryExecutor" }}
+    [celery]
+    default_queue = celery
+{{- end }}
+
+    [scheduler]
+    scheduler_heartbeat_sec = 5
+{{- if .Values.statsd.enabled }}
+    statsd_on = True
+    statsd_port = 9125
+    statsd_prefix = airflow
+    statsd_host = {{ printf "%s-statsd" .Release.Name }}
+{{- end }}
+    # Restart Scheduler every 41460 seconds (11 hours 31 minutes)
+    # The odd time is chosen so it is not always restarting on the same "hour" boundary
+    run_duration = 41460
+
+{{- if .Values.elasticsearch.enabled }}
+    [elasticsearch]
+    # The elasticsearch variables were updated to the shorter names in v1.10.4
+    write_stdout = True
+    json_format = True
+    log_id_template = {dag_id}_{task_id}_{execution_date}_{try_number}
+
+    [elasticsearch_configs]
+    max_retries = 3
+    timeout = 30
+    retry_timeout = True
+{{- end }}
+
+{{- if eq .Values.executor "KubernetesExecutor" }}
+    [kubernetes]
+    namespace = {{ .Release.Namespace }}
+    airflow_configmap = {{ include "airflow_config" . }}
+    airflow_local_settings_configmap = {{ include "airflow_config" . }}
+    worker_container_repository = {{ .Values.images.airflow.repository }}
+    worker_container_tag = {{ .Values.images.airflow.tag }}
+    worker_container_image_pull_policy = {{ .Values.images.airflow.pullPolicy }}
+    worker_service_account_name = {{ .Release.Name }}-worker-serviceaccount
+    image_pull_secrets = {{ template "registry_secret" . }}
+    dags_in_image = True
+    delete_worker_pods = True
+
```

Comment:
@schnie @ashb a follow up question from  this [one](https://github.com/apache/airflow/pull/8777#discussion_r422158586). `.Values.env` can be used to override airflow config and to provide custom env vars for scheduler, webserver and workers when we use the CeleryExecutor , but how is it handled when user opts for `KubernetesExecutor` ? the configmap here doesn't seem to update `[kubernetes_environment_variables]` section of `airflow.cfg` with the custom environment variables set by the user

Reply:
so it's set  as `AIRFLOW__KUBERNETES_ENVIRONMENT_VARIABLES__*` on the scheduler and that will then be used when the worker pod is spawned by the scheduler?

Reply:
It's handled here https://github.com/apache/airflow/pull/8777/files#diff-2af2eefc736b60f59eddbfd5fc655543R70-R79

```

{{- define "custom_airflow_environment" }}
  # Dynamically created environment variables
  {{- range $i, $config := .Values.env }}
  - name: {{ $config.name }}
    value: {{ $config.value | quote }}
    {{- if eq $.Values.executor "KubernetesExecutor" }}
  - name: AIRFLOW__KUBERNETES_ENVIRONMENT_VARIABLES__{{ $config.name }}
    value: {{ $config.value | quote }}
    {{- end }}
  {{- end }}
```

Reply:
What does the `AIRFLOW__KUBERNETES_ENVIRONMENT_VARIABLES__*` do in this instance? Because in the case of setting `airflow_configmap` the environment variable for that setting is just `AIRFLOW__KUBERNETES__AIRFLOW_CONFIGMAP`. So why the need for the extra environment variable?